### DISCUSSION TITLE
Why java 22, and not java 21 (LTS)

### DISCUSSION THREAD

**Discussion by dukeh3** [tcheeric/nostr-java#184] [Ideas]
Hi, I was looking into finding a good java library for nostr to use in my key-manager, and found your (as far as I have seen) excellent library. But why do you use java 22, and not the LTS version java 21? Its a pain to use non LTS version of java in production

**Comment by tcheeric** [comment]
Thanks, @dukeh3. It's mainly because of this: https://github.com/tcheeric/nostr-java/discussions/175

You can easily fork, and modify the corresponding property in the pom file.

**Reply by dukeh3** [reply]
Yes, this is what I did, and it works well for me, (with very very limited testing). But as far as I can see there should be no problem to stop at version 21? As far as I can see you are not using any version 22 functionality?

**Comment by avlo** [comment]
hello, @dukeh3.  please note the repository on which you're commenting ([nostr-java](https://github.com/tcheeric/nostr-java)) uses java 1.8 whereas the java22 variant ([nostr-java-avlo-fork](https://github.com/avlo/nostr-java-avlo-fork)) is a fork of the [nostr-java](https://github.com/tcheeric/nostr-java) 1.8 repo.

that being said, you make a good point re: jdk21 vs jdk22.  if you've already made a jdk21 [nostr-java-avlo-fork](https://github.com/avlo/nostr-java-avlo-fork) variant, you're welcome to submit a pull request to that repo.

otherwise, i'll produce a jdk21 variant of [nostr-java-avlo-fork](https://github.com/avlo/nostr-java-avlo-fork) soon and notify you when it's available.

**Comment by tcheeric** [comment]
I didn't think about that, and wasn't aware of it, when changing to 22 in the latest release. Might have to downgrade again to 21...

**Reply by avlo** [reply]
apologies, eric- i wasn't aware you'd updated nostr-java to jdk22 (didn't get email notification/etc) until reading your above comment.  

fyi, since i'm also downgrading my jdk22 nostr-java variant to jdk21 (given that it's LTS, thx @dukeh3), along with that downgrade are various updates to your framework i'd be happy to submit as a PR if you'd like.

i've got a bit of cleanup to do before then, but the current diff state can be seen [here](https://github.com/tcheeric/nostr-java/compare/develop...avlo:nostr-java-avlo-fork:java21)

**Comment by avlo** [comment]
hi, @tcheeric & @dukeh3.

pom jdk downgrade to v.21 (plus various updates [1](https://github.com/avlo/nostr-java-avlo-fork/commit/904bc3266fbafcf58eb61ddc9911cddad72d3d6f) and [2](https://github.com/avlo/nostr-java-avlo-fork/commit/e8048fd7aa67bdd5dd57054bdec0c21e64dfd7e9) to my fork of eric's develop branch) are available via my repo branch [java21](https://github.com/avlo/nostr-java-avlo-fork/tree/java21).

eric- per your interest, you can examine/comment/critique/merge its diffs as you see fit, or otherwise lmk if you'd prefer i submit a PR to your develop (or other) branch instead.

**Reply by tcheeric** [reply]
Thanks Nick! I'll take a look tonight.

**Comment by dukeh3** [comment]
Ok I have created a PR on my version of nostr-java and downgraded it to java 21.

**Reply by tcheeric** [reply]
Saw that, thanks. I will process later tonight.
